prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,383 @@
+| 2.16.840.1.113762.1.4.1146.798 | Influenza (Tests for influenza A virus Nucleic Acid) | 80588-7 | Influenza virus A M gene [Presence] in Nasopharynx by NAA with probe detection      | LOINC       | 2.74    | Active | Infection caused by novel Influenza A virus variant (disorder) | 541000000000000  | SNOMEDCT              | 2023-03                       | RCTC         |
+| 2.16.840.1.113762.1.4.1146.799 | Influenza (Tests for influenza A virus Antigen)      | 88904-8 | Influenza virus A Ag [Presence] in Lower respiratory specimen by Immunofluorescence | LOINC       | 2.74    | Active | Infection caused by novel Influenza A virus variant (disorder) | 541000000000000  | SNOMEDCT              | 2023-03                       | RCTC         |
+|                                | Influenza - (ABC TESTING LABS)                       | 123456  | Influenza virus A                                                                   | LOCAL       |         | Active | Infection caused by novel Influenza A virus variant (disorder) | 541000000000000  | SNOMEDCT              | 2023-03                       | LOCAL        |   
+
+Example Action Log Error Message: "Missing mapping for: 123456 for table: Observation-Mapping"
+
+In this instance we would add want to add an entry to the Action Log if a null value or empty string was returned. The content of the error message should include both the keyValue and the tableName in the format: "Missing mapping for: " + keyValue + " for table:" + tableName".
+By following this strategy we could also log and monitor missing mappings for any lookupTable which is used in a translation schema, not just condition mappings.
+
+Example Action Log Detail:
+
+```json
+ {
+    "class": "gov.cdc.prime.router.ReportStreamMapping",
+    "scope": "item",
+    "message": "Missing mapping for: 123456 for table: Observation-Mapping",
+    "filteredTrackingElement": "<tracking id>"
+}
+```
+
+## Monitoring Logged Errors
+
+Mapping errors logged by the action log can be monitored through metabase queries by the engagement/customer service team. The following is an example query that would be monitored on a regular cadence (dependent on error volume).
+
+Example query
+```yaml
+select * from action_log
+where type = 'mapping'
+and created_at >= current date -1
+```
+## Unmapped receiver queue
+
+Creation of a receiver with the correct filters will allow us to catch all messages with unmapped observations in a queue for Engagement to work. Combined with Action Log monitoring engagement will be able to update all mappings and resend affected messages as needed to STLTs. In order to create this receiver the following settings would be needed:
```

Comment:
What if one observation was able to get mapped but another wasn't? That message would not end up going here? How do we handle this case?